cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/internal/BootstrapCoordinator.scala

```diff
@@ -212,12 +215,23 @@ private[akka] final class BootstrapCoordinator(discovery: SimpleServiceDiscovery
           if (seedNodes.nonEmpty) {
             log.info("Joining [{}] to existing cluster [{}]", cluster.selfAddress, seedNodes.mkString(", "))
 
-            val seedNodesList = (seedNodes - cluster.selfAddress).toList // order doesn't matter
-            cluster.joinSeedNodes(seedNodesList)
-
-            // once we issued a join bootstrapping is completed
-            replyTo ! BootstrappingCompleted
-            context.stop(self)
+            val addresses = seedNodes - cluster.selfAddress
+
+            // we avoid calling joinSeedNodes with the same set of nodes
+            // this would interrupt and start a new join seed nodes process and cause needless friction
+            // (would still work though)
+            if (addresses != joiningSeedNodesInProgress) {
+              joiningSeedNodesInProgress = addresses
+              cluster.joinSeedNodes(addresses.toList) // order doesn't matter as self node is excluded from the list
+
+              // once we issued a join bootstrapping is completed
+              replyTo ! BootstrappingCompleted
+              context.stop(self)
```

Comment:
something is not matching up here. If it stopped after calling `cluster.joinSeedNodes` then then keeping track of  previous addresses will not help, it's not the same actor instance next time

Looks like `BootstrapCoordinator` actor is started once from `ClusterBootstrap.start`, so something external is retrying the `ClusterBootstrap(system).start()`

Reply:
Indeed, the PR and idea is nonsense I realised once I realised the stop there... 
Need more sleep, will approach from other angle tomorrow